.github/workflows/clp-core-lint.yaml

```diff
@@ -0,0 +1,40 @@
+name: clp-core-lint
+
+on:
+  pull_request:
+    paths:
+      - ".github/workflows/clp-core-lint.yaml"
+      - "components/core/src/**"
+      - "components/core/tests/**"
+  push:
+    paths:
+      - ".github/workflows/clp-core-lint.yaml"
+      - "components/core/src/**"
+      - "components/core/tests/**"
+  workflow_dispatch:
+
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+  # Cancel in-progress jobs for efficiency
+  cancel-in-progress: true
+
+jobs:
+  clang-format:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v3"
+        with:
+          submodules: "recursive"
+
+      - name: "Install and run clang-format on all of core's source files"
+        shell: "bash"
+        working-directory: "./components/core"
+        run: |
+          python3 -m venv venv
```

Comment:
might be unnecessary for now, but I think it could be good in general to explicitly specify the python version like this: https://github.com/y-scope/clp-ffi-py/blob/main/.github/workflows/build_wheels.yml#L46
Advantage: simpler and cleaner Python linter integration if we need them later